persist/fs/seek.go

```diff
@@ -78,6 +78,17 @@ func NewSeeker(filePathPrefix string, bufferSize int, bytesPool pool.BytesPool)
 	}
 }
 
+// GetFileIds retrieves all the identifiers present in the configured file
+func (s *seeker) GetFileIds() []string {
+	fileIds := make([]string, len(s.indexMap))
+	idx := 0
+	for _, indexEntry := range s.indexMap {
+		fileIds[idx] = string(indexEntry.Id)
+		idx++
```

Comment:
To avoid needing the `idx` variable here you can do the following:
```go
ids := make([]string, 0, len(s.indexMap))
for _, entry := range s.indexMap {
	ids = append(ids, ts.BinaryID(entry.Id))
}
return ids
```

This will still avoid allocations as we create the slice with the capacity of what we will insert into it.